BitbucketV2Proxy.cs

```diff
@@ -324,43 +340,110 @@ public string GetOAuthUri(string state = null, string redirectUri = null)
             return sshKeyInfo != null;
         }
 
-        private async Task<BitbucketProxy.BitbucketSSHKeyFullInfo> GetSSHKey(string repoUrl, string accessToken, string sshKey)
+        /// <summary>
+        /// Get Webhook service from v1 API
+        /// </summary>
+        public async Task<BitbucketProxy.BitbucketHookInfo> GetService(string repoUrl, string accessToken, string serviceUrl)
         {
-            var requestUri = BitbucketProxy.GetRequestUri(repoUrl, "deploy-keys");
+            CommonUtils.ValidateNullArgument("repoUrl", repoUrl);
+            CommonUtils.ValidateNullArgument("accessToken", accessToken);
+            CommonUtils.ValidateNullArgument("serviceUrl", serviceUrl);
 
+            var serviceUri = new Uri(serviceUrl);
+            var requestUri = BitbucketProxyHelper.GetRequestUri(APiV1BaseUrl, repoUrl, "services");
             using (var client = CreateHttpClient(accessToken))
+            using (HttpResponseMessage response = await client.GetAsync(requestUri))
+            {
+                BitbucketProxy.BitbucketHookInfo[] services = await ProcessResponse<BitbucketProxy.BitbucketHookInfo[]>("GetService", response);
+                return services.FirstOrDefault(service =>
+                {
+                    if (service.service != null && service.service.url != null && string.Equals(service.service.type, "POST", StringComparison.OrdinalIgnoreCase))
+                    {
+                        Uri configUri;
+                        if (Uri.TryCreate(service.service.url, UriKind.Absolute, out configUri))
+                        {
+                            return string.Equals(serviceUri.Host, configUri.Host, StringComparison.OrdinalIgnoreCase);
+                        }
+                    }
+
+                    return false;
+                });
+            }
+        }
+
+        /// <summary>
+        /// Remove Webhook service from v1 API
+        /// </summary>
+        public async Task<bool> RemoveService(string repoUrl, string accessToken, string serviceUrl)
+        {
+            CommonUtils.ValidateNullArgument("repoUrl", repoUrl);
+            CommonUtils.ValidateNullArgument("accessToken", accessToken);
+            CommonUtils.ValidateNullArgument("serviceUrl", serviceUrl);
+
+            var service = await GetService(repoUrl, accessToken, serviceUrl);
+            if (service != null)
+            {
+                var requestUri = BitbucketProxyHelper.GetRequestUri(APiV1BaseUrl, repoUrl, "services", service.id);
```

Comment:
nope. for services we have to go back to v1 API. 

---
In reply to: [64696631](https://github.com/suwatch/AzureSourceControls/pull/4#discussion_r64696631) [](ancestors = 64696631)